rpcclient/client.go

```diff
@@ -38,24 +38,26 @@ var (
 // running on the local host of a single use system. TLS and authentication are not
 // currently supported.
 type Client struct {
-	ctx      context.Context
-	endpoint string
+	ctx        context.Context
+	endpoint   string
+	wsEndpoint string
 }
 
 // NewClient creates a new JSON-RPC client for the specified endpoint. The passed context
 // is used for the full lifetime of the client.
-func NewClient(ctx context.Context, endpoint string) *Client {
+func NewClient(ctx context.Context, port uint16) *Client {
 	return &Client{
-		ctx:      ctx,
-		endpoint: endpoint,
+		ctx:        ctx,
+		endpoint:   fmt.Sprintf("http://127.0.0.1:%d", port),
+		wsEndpoint: fmt.Sprintf("ws://127.0.0.1:%d/ws", port),
```

Comment:
I didn't want to complicate this PR, but if we want to in the future, I could put the websocket methods at the root too. It is possible to share.